staging/src/k8s.io/apiserver/pkg/server/filters/goaway_test.go

```diff
@@ -348,7 +314,8 @@ func TestClientReceivedGOAWAY(t *testing.T) {
 			// localAddr indicates how many TCP connection set up
 			localAddr := make([]string, 0)
 
-			client, err := newGOAWAYClient(func(network, addr string, cfg *tls.Config) (conn net.Conn, err error) {
+			// create the http client
+			dialFn := func(network, addr string, cfg *tls.Config) (conn net.Conn, err error) {
```

Comment:
`mu` also can be removed I think.

Reply:
`mu` is introduced by https://github.com/kubernetes/kubernetes/pull/94530 try to solve this issue, but did not solve the problem at all I think.

Reply:
yeah, or we add it in both place or we don't add it at all.
I tried adding in both places and with the same result ... so I think that we can remote it